tests/nonbonded/test_nonbonded_interaction_group.py

```diff
@@ -61,7 +61,7 @@ def test_nonbonded_interaction_group_zero_interactions(rng: np.random.Generator)
 
     potential = NonbondedInteractionGroup(num_atoms, ligand_idxs, beta, cutoff)
 
-    du_dx, du_dp, u = potential.to_gpu(np.float64).unbound_impl.execute(conf, params, box)
+    du_dx, du_dp, u = potential.to_gpu(np.float64).unbound_impl.execute(conf, params, box.astype(np.float64))
```

Comment:
what `dtype` does `box = 10.0 * np.eye(3)` produce? (is this a numpy2 change? or a mypy linting thing?)

Reply:
(by default I was under the impression that np.eye(3) is already a `np.float64` in numpy1:

```
>>> import numpy as np
>>> np.eye(3).dtype
dtype('float64')
```

Reply:
Yes, this is to deal with a mypy / numpy typing peculiarity. When a float64 array is multiplied by a Python float, mypy infers the more general dtype `numpy.floating[Any]]` for the result, leading to an error later on when we expect `numpy.float64`:

`example.py`:

```python
from typing import reveal_type

import numpy as np

reveal_type(np.eye(3))
reveal_type(10.0 * np.eye(3))
```

```
$ mypy example.py
tmp.py:5: note: Revealed type is "numpy.ndarray[builtins.tuple[builtins.int, ...], numpy.dtype[numpy.float64]]"
tmp.py:6: note: Revealed type is "numpy.ndarray[builtins.tuple[builtins.int, ...], numpy.dtype[numpy.floating[Any]]]"
Success: no issues found in 1 source file
```